include/pisa/topk_queue.hpp

```diff
@@ -60,14 +60,21 @@ struct topk_queue {
 
     [[nodiscard]] std::vector<entry_type> const& topk() const noexcept { return m_q; }
 
-    void set_threshold(Threshold t) noexcept { m_threshold = t; }
+    void set_threshold(Threshold t) noexcept
+    {
+        m_threshold = std::max(std::nextafter(t, 0.0), 0.0);
+        m_predicted_threshold = t;
+    }
 
     Threshold threshold() const noexcept { return m_threshold; }
 
+    bool is_correct() noexcept { return (m_threshold == m_predicted_threshold); }
```

Comment:
Unnecessary parens and:
```cpp
/// Returns `true` if no documents have been missed up to this point.
/// The reason why document could be missed is forcing a threshold that is too high (overestimated).
bool complete_results() noexcept { return m_threshold == m_predicted_threshold; }
```

Reply:
Unrelated to this PR, but just a curiosity. Can we use `complete_results` to early terminate? :)

Reply:
I don't think so. Now that I'm thinking about it, maybe we should call it `is_safe`.

BTW, shouldn't it be `m_threshold >= m_predicted_threshold`? Or am I missing something?

Reply:
You're right, it should be `>=` - I will fix that.